### ISSUE TITLE
Somfy up/down only trigger one movement step

### ISSUE THREAD

**Issue by Netpoet** [evgeni/esphome-somfy#15]
Thanks for the excellently documented project!
I successfully managed to reproduce it on an ESP32 D1 Mini.

However, when hitting the up/down buttons in the ESPhome web server or the Home Assistant cover component, I seem to be able to trigger nothing but a short movement step.

On the physical remote I trigger this by pressing an up/down button for approximately 3 seconds.
Would you have an idea how to simulate such a long press in the ESPhome cover component?

**Comment by evgeni** [comment]
Moin Frank!

In https://github.com/evgeni/esphome-configs/blob/fa7af8d6faae992c6adb294b7f6821d542e8540b/somfy_cover.h#L35 you should be able to pass a second parameter called `repeat` (defaults to 4) to make the press longer or shorter, depending on your needs!

See also
https://github.com/Legion2/Somfy_Remote_Lib/blob/7b0c2f73401c623dba3ec3d004f547355af234f4/src/SomfyRemote.h#L42
https://github.com/Legion2/Somfy_Remote_Lib/blob/7b0c2f73401c623dba3ec3d004f547355af234f4/src/SomfyRemote.cpp#L13

**Comment by Netpoet** [comment]
Thank you so much, Evgeni! I was too slow to update on my own question. I have actually found the same source.
FYI, I have set the repeating times to 16 now because when set below 13, the signal didn't seem to be received consistantly, and I wanted a few signals of buffer.

The next level will be to group covers and open/close all entities in a group with one button press. Since even the physical remotes don't seem to send signals in a way that reliably open/close a cover completely, this seems to be something we cannot fix in Home Assistant though.

Thank you again for that awesome integration!

**Comment by Netpoet** [comment]
While we are at it - you don't happen to know how to implement another use case: switching three entities at the same time?

I have tried to create a group or a template cover that switches all entities at once. But that doesn't work. I assume Home Assistant triggers overlapping send commands in ESPhome. Instead of building automations with timed delays in Home Assistant, an integration in ESPhome might be more sensible that automatically deploys the trigger signals with the appropriate delays (or a user-defined delay).

Any hints?

**Comment by evgeni** [comment]
Nope, automation with some small delays is the only thing that comes to mind :(

**Comment by freshfieldx** [comment]
Nicely documented and works for without any issues. I wondered if you have looked at a time based cover to allow position to work? I've used this on my Velux in the yaml file successfully, but I can't see how to do the same using the header file.

**Comment by evgeni** [comment]
This is somewhere on the (imaginary) todo list, but I haven't got to it, and we mostly use those blinds as "full open" or "full closed", which the current (ugly) code achieves.

You mention Velux. Which one? 🤔

**Comment by freshfieldx** [comment]
I'm not using the C1101, I've done some very DIY solder at the back of the KLI310 remote! It's not pretty, but works really well to open and close the roof window with automations on Humidity etc.

**Comment by evgeni** [comment]
Ah yeah, the solder method. Thanks!

**Comment by evgeni** [comment]
repeat is configurable since https://github.com/evgeni/esphome-configs/commit/a070515157b69d76e714f9f88dcb7b3fbd3d9eae
